mlrun/api/initial_data.py

```diff
@@ -324,20 +323,37 @@ def _add_data_version(
     db: mlrun.api.db.sqldb.db.SQLDB, db_session: sqlalchemy.orm.Session
 ):
     if db.get_current_data_version(db_session, raise_on_not_found=False) is None:
+        data_version = _get_current_data_version(db, db_session)
```

Comment:
It's a bit weird that in the line above we call `db.get_current_data_version` and when it returns `None` we call 
`_get_current_data_version` (same name, so not clear why it would behave differently)
I would call it `_resolve_current_data_version` but if you have something else your mind, go for it